mvpa2/measures/adhocsearchlightbase.py

```diff
@@ -555,12 +556,25 @@ def assign_ulabels(a):
                   (self.__class__.__name__, time.time() - time_start))
 
         out = Dataset(results)
-        if all_targets:
-            out.sa['targets'] = all_targets
+        self._all_testing_sa = all_testing_sa
         out.sa['cvfolds'] = all_cvfolds
         out.fa['center_ids'] = roi_ids
         return out
 
+    def _pass_attr(self, ds, result):
+        if self._all_testing_sa:
+            all_testing_sa = self._all_testing_sa
+            new_sas = {}
+            for attr in all_testing_sa[0].keys():
+                new_sas[attr] = np.concatenate([tsa[attr].value for tsa in all_testing_sa], 0)
+            result.sa['targets'] = new_sas['targets']
+            mock_attr_ds = Dataset(np.zeros((1,1)))
+            mock_attr_ds.sa = SampleAttributesCollection(new_sas)
+            mock_attr_ds.fa = ds.fa
+            mock_attr_ds.a  = ds.a
+            ds = mock_attr_ds
+        return super(SimpleStatBaseSearchlight, self)._pass_attr(ds, result)
```

Comment:
since iirc there should be no way this was invoked again without redoing training, probably reset that _all_training_sa to None whenever you are done with it here -- why to waste resources?